src/main/java/com/unityTest/testrunner/service/CodeService.java

```diff
@@ -183,4 +199,26 @@ private String buildPythonTestFunction(String funcName, String body) {
 		// Create code block by indenting the body by one tab
 		return def.concat(Utils.indent(body, 1).concat("\n"));
 	}
+
+	private String buildHUnitTestFunction(String funcName, String body) {
+		StringBuilder resultBuilder = new StringBuilder();
+
+		// Importing Hunit
+		resultBuilder.append("import HUnit");
+
+		// Main function
+		resultBuilder.append("\n\nmain :: IO ()\n");
+		resultBuilder.append("main = do\n");
+		resultBuilder.append("  executedTest <- runTestTT test_");
+		resultBuilder.append(funcName);
+		resultBuilder.append("\n");
+		resultBuilder.append("  print executedTest\n\n");
+
+		// Test function
+		resultBuilder.append("test_");
+		resultBuilder.append(funcName);
+		resultBuilder.append(" = ");
+		resultBuilder.append(body);
+		return resultBuilder.toString();
+	}

```

Comment:
This method is too verbose to be called by the `buildTestCaseCode` method. In `buildTestCaseCode` all we want to do it build the bare bones function for testing. You're building what will be displayed when people are looking through test cases written by others. They won't want to see the Hunit import etc. 
I think it should be fine just to do the last bit + the function definition which I'm not seeing here.
```
		resultBuilder.append("test_");
		resultBuilder.append(funcName);
		resultBuilder.append(" = ");
		resultBuilder.append(body);
		return resultBuilder.toString();
```
This should produce a string like what you have in your readme
```
test_{func_name} :: Test
test_{func_name} = {func_body}
```
The rest should go in `writeTestCaseToSuiteFile`, that's where we can do the imports, main function etc.